discovery_engine_core/core/src/storage.rs

```diff
@@ -52,6 +54,26 @@ pub(crate) trait FeedScope {
     async fn store_documents(&self, documents: &[NewDocument]) -> Result<(), Error>;
 }
 
+#[async_trait]
+pub(crate) trait SearchScope {
+    async fn store_new_search(
+        &self,
+        search: &Search,
+        documents: &[NewDocument],
+    ) -> Result<(), Error>;
+
+    async fn clear(&self) -> Result<bool, Error>;
```

Comment:
I'm not sure why I added a boolean there, but I generally think if you call a function to delete something and afterwards it doesn't exist anymore you should not error as it's a success even if nothing was deleted because there is nothing left to delete.

Reply:
i guess that means we should adopt this behavior for the other `clear` methods as well and return a `bool`. i'll open a follow up pr for that.